### PR TITLE
Refactor for version and enum

### PR DESCRIPTION
**Pull request by MikeNeilson**
Sorry for the large amount of code all at once. This will handle the new creation modes as well as it should deal with updating a database.

### CODE REVIEW

**Comment by ktarbet** [comment]
Is there a way to merge this into a Branch within the same project, to make it easier to review in my IDE?

File: src/main/java/hec/TimeSeriesDatabase.java [hunk 1]
```diff
@@ -16,5 +16,5 @@
     public abstract void write(EnsembleTimeSeries ets) throws Exception;	
     public abstract void write(PairedData table);    
     public abstract TimeSeriesIdentifier[] getTimeSeriesIDs();
-    
+    public abstract String getVersion();
```
Comment (HenryGeorgist) [inline src/main/java/hec/TimeSeriesDatabase.java]: good that this is part of the abstraction

File: src/main/java/hec/InvalidCreationMode.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package hec;
+
+public class InvalidCreationMode extends RuntimeException {
```
Comment (HenryGeorgist) [inline src/main/java/hec/InvalidCreationMode.java]: Great idea.

File: src/main/java/hec/JdbcTimeSeriesDatabase.java [hunk 3]
```diff
@@ -23,6 +23,11 @@
  */
 public class JdbcTimeSeriesDatabase extends TimeSeriesDatabase {
 
+    public enum CREATION_MODE {
```
Comment (HenryGeorgist) [inline src/main/java/hec/JdbcTimeSeriesDatabase.java]: is this unique to jdbc?

File: src/main/java/hec/JdbcTimeSeriesDatabase.java [hunk 4]
```diff
@@ -543,4 +591,94 @@ public String getSQLTableName(String catalog_name, String type) throws SQLExcept
...
+            while( (line = br.readLine()) != null ){
+                sb.append(line).append(System.lineSeparator());
+            }
+            return sb.toString();
+        }catch( IOException err ){
+            throw new RuntimeException("Error extracting defined resource information",err);
+        } 
+	}    
+
+    private void updateFor20200101_to_20200224() {
```
Comment (HenryGeorgist) [inline src/main/java/hec/JdbcTimeSeriesDatabase.java]: thanks.

File: src/test/java/hec/DatabaseVersioning.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        try( TimeSeriesDatabase db = new JdbcTimeSeriesDatabase(testdb.getCanonicalPath(), JdbcTimeSeriesDatabase.CREATION_MODE.OPEN_EXISTING_UPDATE);){
+            assertEquals( get_latest_version(db.getVersions()), db.getVersion() );
+        }
+        
+        testdb.delete();
+    }
+
+
+
+    public File get_test_file( String test_database_resource ) throws Exception{
```
Comment (ktarbet) [inline src/test/java/hec/DatabaseVersioning.java]: can we make get_test_file  private, and the next.

File: src/main/java/hec/JdbcTimeSeriesDatabase.java (hunk 3 above)
Comment (MikeNeilson) [reply src/main/java/hec/JdbcTimeSeriesDatabase.java]: It's more specific to SQLite but we called it JDBC. If we were use library forfor say, the  CWMS oracle database, there wouldn't be any option except open the existing.

File: src/main/java/hec/JdbcTimeSeriesDatabase.java [hunk 6]
```diff
@@ -38,16 +43,47 @@
      *                 exist)
      * @throws Exception
      */
-    public JdbcTimeSeriesDatabase(String database, boolean create) throws Exception {
+    public JdbcTimeSeriesDatabase(String database, CREATION_MODE creation_mode) throws Exception {
```
[the remark this answers is not in the document]
Comment (MikeNeilson) [reply src/main/java/hec/JdbcTimeSeriesDatabase.java]: Not that I can think of as of yet. But moment one of us thinks we have we should submit code and argue, I mean civilly discuss, about it.

File: src/test/java/hec/DatabaseVersioning.java (hunk 5 above)
Comment (MikeNeilson) [reply src/test/java/hec/DatabaseVersioning.java]: yes, that shouldn't be an issue.

File: src/main/java/hec/JdbcTimeSeriesDatabase.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (HenryGeorgist) [reply src/main/java/hec/JdbcTimeSeriesDatabase.java]: I say we merge in, use and learn.

**Pull request merged**